feature/floor-map/src/main/java/io/github/droidkaigi/confsched2023/floormap/FloorMapScreen.kt

```diff
@@ -115,42 +141,123 @@ private fun FloorMapScreen(
-                Modifier
-                    .padding(innerPadding)
-                    .padding(horizontal = 16.dp),
-            ) {
-                Column(
-                    Modifier
-                        .fillMaxSize(),
-                ) {
-                    FloorMap(uiState = uiState.floorMapUiState)
-                    FloorMapSideEventList(
-                        uiState = uiState.floorMapSideEventListUiState,
-                        onSideEventClick = onSideEventClick,
-                        modifier = Modifier
-                            .weight(1f)
-                            .padding(bottom = 56.dp)
-                            .fadingEdge(
-                                Brush.verticalGradient(
-                                    0.85f to Color.Black,
-                                    1f to Color.Transparent,
-                                ),
-                            ),
-                    )
-                }
-                FloorLevelSwitcher(
-                    selectingFloorLevel = uiState.floorLevel,
-                    onClickFloorLevelSwitcher = onClickFloorLevelSwitcher,
-                    modifier = Modifier
-                        .align(Alignment.BottomCenter)
-                        .padding(bottom = 24.dp),
-                )
-            }
+            FloorMapContent(
+                innerPadding = innerPadding,
+                widthSizeClass = widthSizeClass,
+                uiState = uiState,
+                onSideEventClick = onSideEventClick,
+                onClickFloorLevelSwitcher = onClickFloorLevelSwitcher,
+            )
         },
     )
 }
 
+@Composable
+private fun FloorMapContent(
+    innerPadding: PaddingValues,
+    widthSizeClass: WindowWidthSizeClass,
+    uiState: FloorMapScreenUiState,
+    onSideEventClick: (url: String) -> Unit,
+    onClickFloorLevelSwitcher: (FloorLevel) -> Unit,
+) {
+    BoxWithConstraints(
+        Modifier
+            .padding(innerPadding)
+            .padding(horizontal = 16.dp),
+    ) {
+        if (widthSizeClass != WindowWidthSizeClass.Compact) {
+            LargeScreenContent(
+                uiState = uiState,
+                onSideEventClick = onSideEventClick,
+            )
+        } else {
+            MobileContent(
+                uiState = uiState,
+                onSideEventClick = onSideEventClick,
+            )
+            FloorLevelSwitcher(
+                selectingFloorLevel = uiState.floorLevel,
+                onClickFloorLevelSwitcher = onClickFloorLevelSwitcher,
+                modifier = Modifier
+                    .align(Alignment.BottomCenter)
+                    .padding(bottom = 24.dp),
+            )
+        }
+    }
+}
+
+@Composable
+private fun MobileContent(
+    uiState: FloorMapScreenUiState,
+    onSideEventClick: (url: String) -> Unit,
+) {
+    Column(
+        modifier = Modifier.fillMaxSize(),
+    ) {
+        FloorMap(uiState = uiState.floorMapUiState)
```

Comment:
How about having floorMapUiState in smallFloorMapContentUiState and make smallFloorMapContentUiState this composable parameter?

Reply:
surely.
LargeFloorMapContentUiState and SmallFloorMapContentUiState should have floorMapUiState.
I'll change that.

Reply:
fixed [Changed the way you hold data](https://github.com/DroidKaigi/conference-app-2023/pull/814/commits/ef212fac6f81e40ad5dd3866cf88bbd240c7e835)